src/packet_analysis/protocol/vlan/VLAN.cc

```diff
@@ -17,19 +17,13 @@ bool VLANAnalyzer::AnalyzePacket(size_t len, const uint8_t* data, Packet* packet
     }
 
     uint16_t tci = (data[0] << 8u) + data[1];
-    auto vlan_id = tci & 0xfff;
-    auto vlan_pcp = (tci & 0xe000) >> 13;
+    uint16_t vlan_id = tci & 0xfff;
+    uint8_t vlan_pcp = (tci & 0xe000) >> 13;
     bool vlan_dei = (tci & 0x1000) != 0;
-    if ( packet->vlan == 0 ) {
-        packet->vlan = vlan_id;
-        packet->vlan_pcp = vlan_pcp;
-        packet->vlan_dei = vlan_dei;
-    }
-    else {
-        packet->inner_vlan = vlan_id;
-        packet->inner_vlan_pcp = vlan_pcp;
-        packet->inner_vlan_dei = vlan_dei;
-    }
+    if ( ! packet->GetVlanTag() )
+        packet->SetVlanTag({.id = vlan_id, .pcp = vlan_pcp, .dei = vlan_dei});
+    else
+        packet->SetInnerVlanTag({.id = vlan_id, .pcp = vlan_pcp, .dei = vlan_dei});
```

Comment:
Definitely existing, I wonder if we should check for InnerVlanTag() being already set (no need to do it in this PR)
```
if ( GetInnerVlanTag() )
    reporter->Weird("triple_tagged_vlan_qinqinq_unsupported");
```

Reply:
maybe, I mean really you can have arbitrarily nested vlan tags, right? So `triple` would be a misnomer in there, and really Zeek *should* handle that. Not sure why we only handle the outermost/innermost, but it's pretty clear looking at Zeek's documentation that that's the expected behavior.

I would instead favor changing this to a vector or something to handle arbitrary numbers of VLAN tags.